File: uva_455.py
def check(s, k):
    for i in range(k):
        j=k
        while i+j<len(s):
            if s[i]!=s[i+j]:
                return False
            j+=k
    return True

num=int(input())
while num:
    num-=1
    data=input()
    data=input()
    for i in range(1, len(data)+1):
        if not len(data)%i:
            if check(data, i):
                print(i)
                if num:
                    print('')
                break

File: uva_455.cpp
# include <string>
# include <iostream>
using namespace std;

bool check(string s, int k){
    for (int i=0; i<k; ++i)
        for (int j=k; i+j<s.length(); j+=k)
            if (s[i]!=s[i+j])
                return false;
    return true;
}
int main(void){
    int num;
    cin >> num;
    while(num--){
        string data="";
        cin >> data;
        for (int i=1; i<=data.length(); ++i){
            if ((data.length()%i)==0)
                if (check(data, i)){
                    cout << i << endl;
                    if (num)
                        cout << endl;
                    break;
                }
        }
    }
    return 0;
}